pkg/generators/forwarding/fluentd/output_conf_forward_test.go

```diff
@@ -0,0 +1,125 @@
+package fluentd
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	test "github.com/openshift/cluster-logging-operator/test"
+)
+
+var _ = Describe("Generating fluentd secure forward output store config blocks", func() {
+
+	var (
+		outputs   []logging.OutputSpec
+		generator *ConfigGenerator
+	)
+	BeforeEach(func() {
+		var err error
+		generator, err = NewConfigGenerator()
+		Expect(err).To(BeNil())
+	})
+
+	Context("for a secure endpoint", func() {
+		BeforeEach(func() {
+			outputs = []logging.OutputSpec{
+				{
+					Type:     logging.OutputTypeForward,
+					Name:     "secureforward-receiver",
+					Endpoint: "es.svc.messaging.cluster.local:9654",
+					Secret: &logging.OutputSecretSpec{
+						Name: "my-infra-secret",
+					},
+				},
+			}
+		})
+
+		It("should produce well formed output label config", func() {
+			results, err := generator.generateOutputLabelBlocks(outputs)
+			Expect(err).To(BeNil())
+			Expect(len(results)).To(Equal(1))
+			test.Expect(results[0]).ToEqual(`<label @SECUREFORWARD_RECEIVER>
+	<match **>
+		# https://docs.fluentd.org/v1.0/articles/in_forward
+	   @type forward
+	   <security>
+	     self_hostname "#{ENV['NODE_NAME'] || ${hostname}}" 
```

Comment:
This seems to be causing issues with fluentd:

```
/opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/literal_parser.rb:171:in `instance_eval': (eval):3: `${' is not allowed as a global variable name (SyntaxError)
(eval):3: syntax error, unexpected end-of-input
"#{ENV['NODE_NAME'] || ${hostname}}"
                       ^
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/literal_parser.rb:171:in `eval_embedded_code'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/literal_parser.rb:93:in `scan_double_quoted_string'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/literal_parser.rb:72:in `scan_string'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/literal_parser.rb:65:in `parse_literal'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:130:in `parse_element'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:95:in `parse_element'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:43:in `parse!'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config/v1_parser.rb:33:in `parse'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/config.rb:39:in `parse'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/supervisor.rb:775:in `read_config'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/supervisor.rb:531:in `run_worker'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/lib/fluent/command/fluentd.rb:320:in `<top (required)>'
        from /opt/rh/rh-ruby25/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /opt/rh/rh-ruby25/root/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:59:in `require'
        from /opt/rh/rh-ruby25/root/usr/local/share/gems/gems/fluentd-1.7.3/bin/fluentd:8:in `<top (required)>'
        from /opt/rh/rh-ruby25/root/usr/local/bin/fluentd:23:in `load'
        from /opt/rh/rh-ruby25/root/usr/local/bin/fluentd:23:in `<main>'
```

Reply:
ill remove it.  I was trying to configure it like it was previously.